File: src-tauri/src/models/house.rs
use crate::schema::houses;
use diesel::{Identifiable, Queryable, Selectable};
use serde::Serialize;

use super::apartment::Apartment;

#[derive(Identifiable, Queryable, Selectable, Serialize, Debug, PartialEq, Eq)]
#[diesel(table_name = houses)]
#[diesel(primary_key(house_name))]
pub struct House {
    pub house_name: String,
    pub street_type: String,
    pub street_name: String,
    pub street_number: i32,
}

#[derive(Serialize)]
pub struct HouseWithApartments {
    pub house: House,
    pub apartments: Vec<Apartment>,
}

File: src/lib/models/house.ts
import { z } from "zod"

import { Constants } from "$lib/utils/constants"

import { ApartmentSchema } from "./apartment"

export const HouseSchema = z.object({
  house_name: z.string().regex(Constants.houseRegex),
  street_type: z.string(),
  street_name: z.string(),
  street_number: z.number()
})

export type House = z.infer<typeof HouseSchema>

export const HouseWithApartmentsSchema = z.object({
  house: HouseSchema,
  apartments: z.array(ApartmentSchema)
})

export type HouseWithApartments = z.infer<typeof HouseWithApartmentsSchema>
